kong/runloop/balancer.lua

```diff
@@ -752,6 +752,16 @@ end
 -- which do not support sending it)
 -- @return true on success, nil+error message+status code otherwise
 local function execute(target, ctx)
+  do -- Check for KONG_ORIGINS override
+    local origin_key = target.scheme .. "://" .. utils.format_host(target)
```

Comment:
Since this comparison might be expensive maybe cache the results?

Does this cater for matching different formats? casing in names, shortening in ipv6? could use some accompanying tests

Reply:
What do you mean "comparison" here? What would you cache?

Reply:
sorry bad wording. Creating the key and then looking it up (=comparison) in the `origins` table.

so if `origins` has

```lua
{
  ["http://[::1]:80"] = {
    scheme = "http",
    host = "somewhere.else",
    port = 80,
  }
}
```

and incoming is `http://[0000::1]:80` then it doesn't match, so you need to normalize. That is expensive, and that is something you could cache.